lib/neo4j/active_node/dependent/association_methods.rb

```diff
@@ -28,7 +28,8 @@ def dependent_delete_callback(object)
         end
 
         def dependent_delete_orphans_callback(object)
-          object.as(:self).unique_nodes(self, :self, :n, :other_rel).query.delete(:n, :other_rel).exec
+          unique_query = object.as(:self).unique_nodes(self, :self, :n, :other_rel)
+          unique_query.query.detach_delete(:n).exec if unique_query
```

Comment:
Is truthiness of `unique_query` enough to prove that it should execute the query?

Reply:
A couple of things I'm looking into (again, sorry for the delay in looking at this):

 * We can't use `detach_delete` because we need to support versions of Neo4j before 3.0
 * `unique_nodes` returns false for the case of `send(association.name).empty?`, otherwise is returns a `QueryProxy` (which would always be considered truthy).  I'm actually a bit puzzled because before if `unique_nodes` returned false we wouldn't be able to call `query` on it, so that situation was probably never coming up in the tests.  But if that's the case, I don't yet understand how the result of `unique_query` could not be truthy.

Still digging on that second one...

Reply:
Ah, I see now that the relationship variable in the `delete` was the problem.  Cool, continuing to look...